### PR TITLE
Initial separation of concerns

### PR DESCRIPTION
**Pull request by KonstancjaMG**

### CODE REVIEW

File: js/pages/students.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  </div>
+</div>
+
+  `
+}
+
+function createStudentCards(pStudentsArray) {
+  const cards = pStudentsArray.map(student => {
+      return `
+        <div class="col-12 col-md-6 col-lg-6 col-xl-4 mb-4">
```
Comment (mehmet-hicoders) [inline js/pages/students.js:53]: You can extract this templates to another file

File: js/render.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
 import { renderNav } from "./navigation.js";
-import {createSeparator} from "./spacer.js";
-import { renderContent } from "./content/content.js";
+import { renderContent } from "./pages/content.js";
+import { renderHome } from "./pages/home/home.page.js";
 
 const renderApp = () => {
     renderNav();
-    createSeparator();
     renderContent();
```
Comment (mehmet-hicoders) [inline js/render.js:7]: I think "js/pages/content.js" renderContent() is already handling creation of contentWrapper.
Having this line here is not good.
I would create a dom.js and create a function like "getContentContainer" and all renderXYZ content function imports this function to access the contenetcontainer.

File: js/event_listeners/students/search.event.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            cardsContainer.innerHTML = createStudentCards(filteredStudents);
+        }
+    });
+}
+
+function searchStudentsByContactNr() {
+    const searchInput = document.getElementById('students-search-contact');
+    const cardsContainer = document.getElementById('cardsColumn');
+    searchInput.addEventListener('input', function () {
+        const studentsArray = loadFromLocalStorage('students');
```
Comment (mehmet-hicoders) [inline js/event_listeners/students/search.event.js:103]: Put this line after line 101!
You do not need to read students strucutre for each user entry!
What would happen if you would have 10000 students in storage?

File: js/event_listeners/students/add.event.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        const dobInput = document.getElementById('students-add-dob');
+        const placeInput = document.getElementById('students-add-place');
+        const contactInput = document.getElementById('students-add-contact');
+        const addButton = document.getElementById('add-btn');
+
+        [nameInput, surnameInput, classInput, dobInput, placeInput, contactInput].forEach(input => {
+            input.classList.remove('border-danger');
+        });
+
+        const name = document.getElementById('students-add-name').value;
```
Comment (mehmet-hicoders) [inline js/event_listeners/students/add.event.js:39]: You already access this element in line 27

File: js/event_listeners/students/add.event.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    address,
+    parentsContact: contact
+});
+
+
+
+function addStudent() {
+    const addBtn = document.getElementById('add-btn');
+
+    addBtn.addEventListener('click', function() {
```
Comment (mehmet-hicoders) [inline js/event_listeners/students/add.event.js:26]: This function is too long. Try to have your functions with maximum 11 lines

**Review by mehmet-hicoders — CHANGES_REQUESTED** [review]

File: js/pages/students.js (hunk 1 above)
Comment (KonstancjaMG) [reply js/pages/students.js:53]: implemented in the next, newer branch

File: js/render.js (hunk 2 above)
Comment (KonstancjaMG) [reply js/render.js:7]: resolved in the newer PR

File: js/event_listeners/students/search.event.js (hunk 3 above)
Comment (KonstancjaMG) [reply js/event_listeners/students/search.event.js:103]: restructurized to a single searchfunction with a debounce in the newest branch

File: js/event_listeners/students/add.event.js (hunk 5 above)
Comment (KonstancjaMG) [reply js/event_listeners/students/add.event.js:26]: Again, implemented in a newer branch but here is a preview. A couple lines more than 11 but I think it's a success.

```
function addStudent() {
    const inputs = getInputFields();

    inputs.addButton.addEventListener('click', function() {
        resetInputBorders(inputs);

        const studentData = getStudentData(inputs);
        if (isInputDataValid(studentData)) {
            addNewStudentToLocalStorage(studentData);
            renderStudents();
            showSuccessMessage('Student added successfully!');
        } else {
            addBorderIfInvalid(inputs, studentData);
        }
    });
}

```

**Pull request closed without merging**